test: migrate `FeeHandlerSeller.t.sol` to foundry

arthurgousset opened the pull request:
### Description

1. adds `FeeHanderSeller.t.sol` (migrates all unit tests from `feehandlerseller.ts` to Foundry)
2. removes `feehandlerseller.ts` (fully covered by Foundry)

### Other changes

None

### Tested

Yes, both locally and on CI ✅ 

```sh
forge test -vvvv \
--match-path "test-sol/unit/common/FeeHandlerSeller.t.sol"
```

### Related issues

- Fixes #10802
- Related to #11150

### Backwards compatibility

No, this removes TS tests, in favour of new Foundry tests.



### Documentation

No

arthurgousset committed:
test(FeeHandlerSeller.t.sol): adds MVP contract outline

Test is currently failing

arthurgousset committed:
test(FeeHandlerSeller.t.sol): use `GoldTokenMock` to work around missing transfer precompile

Test passes as expected ✅

arthurgousset committed:
chore(FeeHandlerSeller.t.sol): remove unused code comments

arthurgousset committed:
test(FeeHandlerSeller.t.sol): add test for revert when not owner

Test passes as expected ✅

arthurgousset committed:
style(FeeHandlerSeller.t.sol): improve function naming

More explicit naming in compliance with VS code extension that suggests forge test naming.

arthurgousset committed:
test(FeeHandlerSeller.t.sol): add test for setMinimumReports

Test passes as expected ✅

arthurgousset committed:
test(FeeHandlerSeller.t.sol): refactor constants and simplify tests

All tests pass as expected ✅

```sh
forge test -vvvv \
--match-path "test-sol/unit/common/FeeHandlerSeller.t.sol"
```

arthurgousset committed:
test(FeeHandlerSeller.t.sol): remove unused imports

All tests pass as expected ✅

```sh
forge test -vvvv \
--match-path "test-sol/unit/common/FeeHandlerSeller.t.sol"
```

arthurgousset committed:
test(feehandlerseller.ts): deletes TS test

Fully migrated to foundry test.

martinvol commented on packages/protocol/test-sol/unit/common/FeeHandlerSeller.t.sol:
```diff
@@ -0,0 +1,127 @@
+}
+
+contract FeeHandlerSellerTest_SetMinimumReports is FeeHandlerSellerTest {
+  address ARBITRARY_TOKEN_ADDRESS = MOCK_CELO_TOKEN_ADDRESS;
```
nit: would be good to get a random address that's different than the Celo one, just in case

arthurgousset replied:
Great point, that's done in https://github.com/celo-org/celo-monorepo/pull/11155/commits/3af706f5483bb3a2ca01d939c668f8b272b0f395

arthurgousset committed:
refactor(FeeHandlerSeller.t.sol): use arbitrary token address

All tests pass as expected ✅

```sh
forge test -vvvv \
--match-path "test-sol/unit/common/FeeHandlerSeller.t.sol"
```
